File: src/api/log.ts
import { invoke } from '@tauri-apps/api/core';

export async function trace(message: string): Promise<void> {
  await invoke('trace', { message });
}

export async function debug(message: string): Promise<void> {
  await invoke('debug', { message });
}

export async function info(message: string): Promise<void> {
  await invoke('info', { message });
}

export async function warn(message: string): Promise<void> {
  await invoke('warn', { message });
}

export async function error(message: string): Promise<void> {
  await invoke('error', { message });
}

File: src-tauri/src/api/log.rs
#[tauri::command]
pub fn trace(message: String) {
    spdlog::trace!("{}", message);
}

#[tauri::command]
pub fn debug(message: String) {
    spdlog::debug!("{}", message);
}

#[tauri::command]
pub fn info(message: String) {
    spdlog::info!("{}", message);
}

#[tauri::command]
pub fn warn(message: String) {
    spdlog::warn!("{}", message);
}

#[tauri::command]
pub fn error(message: String) {
    spdlog::error!("{}", message);
}
